roles/nfd_operator_check_resource_statuses/vars/main/resources.yml

```diff
@@ -0,0 +1,41 @@
+---
```

Comment:
I fear that this list is too specific to the current version of NFD, and it would be hard to maintain it and keep it in sync with changes in the NFD operator. Once something changes in the NFD operator, we would have to update this list and also have to maintain backwards compatibility, because we might be testing both older and newer versions of the NFD operator with the same ci-artifacts version.

To overcome this, I suggest that instead of hard-coding the names of the resources, maybe it would be best to just dump all resources of each kind in the `openshift-nfd` namespace.

For cluster-scoped resources such as the namespace itself, the SCC and the ClusterRoles, which are not part of the `openshift-nfd` namespace, we can hardcode for now. But this too can be solved if the nfd-operator would label its cluster-scoped resources with some label (like `"nodefeaturediscoveried.nfd.openshinshift.io-v1-resource": "true"` or something). Then we can just scan for all cluster-scoped resources with that label and dump them.

Reply:
Thanks Omer! 

That makes a lot of sense -- we don't want to have difficulties in the future with maintaining the NFD Operator.

@ArangoGutierrez - what do you think about Omer's cluster-scoped resource labels suggestion? I think his suggestion would be useful to add to the NFD Operator.

Reply:
> To overcome this, I suggest that instead of hard-coding the names of the resources, maybe it would be best to just iterate on all resources of each kind in the openshift-nfd namespace.

@courtneypacheco , I agree with @omertuc that you should try to avoid hard-coding the names of the resources
```
 # If no $resource were found at all, then something is wrong.
 # Attempt to retreive the $resource from the cluster to see if they exist
 # If any of the NFD Operator $resource do not exist, then we cannot continue
```

most of the files follow this pattern, I think there is a way for Ansible to move the resource type to a variable, something like
```

- name: Attempt to retreive the NFD Operator {{ resource }}
  shell: |
    set -o pipefail;
    oc get {{ resource }} '{{ resource_name.stdout }}' -n '{{ nfd_operator_namespace }}' -oname | grep 'error'
```
and maybe at the end of the generic file, `include` the resource-specific  file for further testing, if it exists

Reply:
Agree with @omertuc , `oc get all` on the `openshift-nfd` NS should be enough resources to chaos test NFD and then see if the operator is checking the status of the given objects